classes/task/crawl_manager.php

```diff
@@ -0,0 +1,64 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * tool_crawler
+ *
+ * @package    tool_crawler
+ * @copyright  2016 Brendan Heywood <brendan@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_crawler\task;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once("$CFG->dirroot/admin/tool/crawler/lib.php");
+
+/**
+ * crawl_manager
+ * Creates a batch of crawls to be done in an ad hoc task.
+ * This gives us control over how parallel it is, and also when the workload is processed (eg at night).
+ *
+ * @package    tool_crawler
+ * @copyright  2016 Brendan Heywood <brendan@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class crawl_manager extends \core\task\scheduled_task {
```

Comment:
you still have the old normal cron task:

https://github.com/catalyst/moodle-tool_crawler/blob/master/classes/task/crawl_task.php#L39

So in practice currently we will have that cron task as well as the N ad hoc tasks. Is this intended?

Reply:
I thought we may want to keep the old method, but it doesn't make sense. They both just call the tool_crawler_crawl method. We can have the adhoc task running with a parallel setting of 1 and achieve the same thing.

Reply:
Given that they are so similar, maybe we should not rename it but just re-purpose the existing shceduled task?

Reply:
Makes sense, I've made a change to repurpose the existing scheduled task instead